src/coreclr/inc/corinfo.h

```diff
@@ -1553,6 +1553,11 @@ struct CORINFO_RESOLVED_TOKEN
     uint32_t                cbTypeSpec;
     PCCOR_SIGNATURE         pMethodSpec;
     uint32_t                cbMethodSpec;
+
+    //
+    // [Metadata] If the token refers to a constrained call, the token of the constraint
+    //
+    mdToken                 constrainedToken;
```

Comment:
This needs a matching change to `Agnostic_CORINFO_RESOLVED_TOKENin` and methods using it for mch collection.

Reply:
Is this really an `in` parameter? The change as-is will currently only populate this _after_ calling `resolveToken`. I'm not sure I really understand how SPMI operates but it feels like RyuJIT should be able to populate this field on its own without SPMI help.

Reply:
If I read the change right the implementation of `getCallInfo` is using the information to make decisions. That means SPMI needs to key on this information to replay `getCallInfo` correctly, which I think will only happen correctly if you put it in `Agnostic_CORINFO_RESOLVED_TOKENin`.
It will also cause `resolveToken` to key on it, but that should be ok if it is always 0 there.

Reply:
Ah sorry, I just noticed that `getReadyToRunDelegateCtorHelper` is the function that uses the new information. In that case it is the same point as above -- when SPMI replays `getReadyToRunDelegateCtorHelper` it needs to key on this information to replay it correctly.

Reply:
Ok, done. It's a bit of a misuse but I couldn't find a better spot to transfer this information within RyuJIT.

We don't really _need_ it on `CORINFO_RESOLVED_TOKEN` (we could just introduce a new argument to `getReadyToRunDelegateCtorHelper` to communicate this back to the EE side, since we're changing the JitInterface anyway), but I didn't see a better way to attach this information to RyuJIT's data structures.